test/openssl/test_ssl_session.rb

```diff
@@ -239,10 +242,12 @@ def test_ctx_client_session_cb_tls12
       end
 
       server_connect_with_session(port, ctx, nil) { |ssl|
-        assert_equal(1, ctx.session_cache_stats[:cache_num])
+        # AWS-LC doesn't support internal session caching on the client, but
+        # the callback is still enabled as expected.
+        assert_equal(1, ctx.session_cache_stats[:cache_num]) if !aws_lc?
         assert_equal(1, ctx.session_cache_stats[:connect_good])
         assert_equal([ssl, ssl.session], called[:new])
-        assert_equal(true, ctx.session_remove(ssl.session))
+        assert_equal(true, ctx.session_remove(ssl.session)) if !aws_lc?
         assert_equal(false, ctx.session_remove(ssl.session))
         if TEST_SESSION_REMOVE_CB
           assert_equal([ctx, ssl.session], called[:remove])
```

Comment:
The part inside `if TEST_SESSION_REMOVE_CB` also needs to be skipped.

I think putting all assertions related to the internal cache in a multiline `if/unless-end` would make this clearer.